src/measure_theory/function/uniform_integrable.lean

```diff
@@ -636,4 +640,199 @@ lemma tendsto_in_measure_iff_tendsto_Lp [is_finite_measure μ] (hp : 1 ≤ p) (h
 
 end unif_integrable
 
+section uniform_integrable
+
+/-! `uniform_integrable`
+
+In probability theory, uniform integrability normally refers to the condition that a sequence
+of function `(fₙ)` satisfies for all `ε > 0`, there exists some `C ≥ 0` such that
+`∫ x in {|fₙ| ≥ C}, fₙ x ∂μ ≤ ε` for all `n`.
+
+In this section, we will develope some API for `uniform_integrable` and prove that
+`uniform_integrable` is equivalent to this definition of uniform integrability.
+(Currently we only have the forward direction.)
+-/
+
+variables [measurable_space β] [hβ : second_countable_topology β]
+  {p : ℝ≥0∞} {f : ι → α → β}
+
+lemma uniform_integrable_zero_meas [measurable_space α] {p : ℝ≥0∞}
+  {f : ι → α → β} (hf : ∀ i, measurable (f i)) :
+  uniform_integrable f p (0 : measure α) :=
+⟨hf, unif_integrable_zero_meas, 0, λ i, snorm_measure_zero.le⟩
+
+lemma uniform_integrable_of_ae_eq {p : ℝ≥0∞} {f g : ι → α → β}
+  (hf : uniform_integrable f p μ) (hg : ∀ i, measurable (g i)) (hfg : ∀ n, f n =ᵐ[μ] g n) :
+  uniform_integrable g p μ :=
+begin
+  obtain ⟨-, hunif, C, hC⟩ := hf,
+  refine ⟨hg, (unif_integrable_congr_ae hfg).1 hunif, C, λ i, _⟩,
+  rw ← snorm_congr_ae (hfg i),
+  exact hC i
+end
+
+lemma uniform_integrable_congr_ae {p : ℝ≥0∞} {f g : ι → α → β}
+  (hf : ∀ i, measurable (f i)) (hg : ∀ i, measurable (g i)) (hfg : ∀ n, f n =ᵐ[μ] g n) :
+  uniform_integrable f p μ ↔ uniform_integrable g p μ :=
+⟨λ h, uniform_integrable_of_ae_eq h hg hfg,
+ λ h, uniform_integrable_of_ae_eq h hf (λ i, (hfg i).symm)⟩
+
+variables [borel_space β]
+
+/-- A finite sequence of Lp functions is uniformly integrable in the probability sense. -/
+lemma uniform_integrable_fintype [fintype ι] (hp_one : 1 ≤ p) (hp_top : p ≠ ∞)
+  {f : ι → α → β} (hf : ∀ i, measurable (f i)) (hf' : ∀ i, mem_ℒp (f i) p μ) :
+  uniform_integrable f p μ :=
+begin
+  refine ⟨hf, unif_integrable_fintype μ hp_one hp_top hf', _⟩,
+  by_cases hι : nonempty ι,
+  { choose ae_meas hf using hf',
+    set C := (finset.univ.image (λ i : ι, snorm (f i) p μ)).max'
+      ⟨snorm (f hι.some) p μ, finset.mem_image.2 ⟨hι.some, finset.mem_univ _, rfl⟩⟩,
+    refine ⟨C.to_nnreal, λ i, _⟩,
+    rw ennreal.coe_to_nnreal,
+    { exact finset.le_max' _ _ (finset.mem_image.2 ⟨i, finset.mem_univ _, rfl⟩) },
+    { refine ne_of_lt ((finset.max'_lt_iff _ _).2 (λ y hy, _)),
+      rw finset.mem_image at hy,
+      obtain ⟨i, -, rfl⟩ := hy,
+      exact hf i } },
+  { exact ⟨0, λ i, false.elim $ hι $ nonempty.intro i⟩ }
+end
+
+/-- A single function is uniformly integrable in the probability sense. -/
+lemma uniform_integrable_subsingleton [subsingleton ι] (hp_one : 1 ≤ p) (hp_top : p ≠ ∞)
+  {f : ι → α → β} (hf : ∀ i, measurable (f i)) (hf' : ∀ i, mem_ℒp (f i) p μ) :
+  uniform_integrable f p μ :=
+uniform_integrable_fintype hp_one hp_top hf hf'
```

Comment:
Can you also add a lemma `uniform_integrable_const`, which would state that a single function is uniformly integrable, this time represented as a constant sequence of functions? That is, `uniform_integrable (λ i, f) p µ`, with `f : α → β`.